### PR TITLE
fix(sql): fix like/ilike filter on bind variable expression

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Fixes "use constant or bind variable" error returned by queries with expressions like `ILIKE '%' || :str || '%'` in the filter.

Also fixes flaky `O3FailureTest` (`Overrides` weren't reset due to missing `tearDown()` call).

### CODE REVIEW

File: core/src/test/java/io/questdb/test/griffin/engine/functions/regex/ILikeFunctionFactoryTest.java [hunk 1]
```diff
@@ -26,15 +26,31 @@
...
 import io.questdb.cairo.sql.RecordCursorFactory;
-import io.questdb.test.AbstractGriffinTest;
 import io.questdb.griffin.SqlException;
 import io.questdb.std.Chars;
+import io.questdb.test.AbstractGriffinTest;
 import io.questdb.test.tools.TestUtils;
 import org.junit.Assert;
 import org.junit.Test;
 
 public class ILikeFunctionFactoryTest extends AbstractGriffinTest {
```
Comment (bziobrowski) [inline core/src/test/java/io/questdb/test/griffin/engine/functions/regex/ILikeFunctionFactoryTest.java:37]: If test is meant to mimic pg-wire behavior with bind vars then it'd be a bit better to use indexed bind variable .

Comment (puzpuzpuz) [reply core/src/test/java/io/questdb/test/griffin/engine/functions/regex/ILikeFunctionFactoryTest.java:37]: The goal is to cover behavior with bind variable expression for all possible cases. Added missing tests for indexed vars in bd5e0adc1191def83438f8452463f0e139aa1745

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 3 (66.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bind/UuidBindVariable.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/regex/AbstractLikeStrFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bind/NamedParameterLinkFunction.java|1|1|100.00%|

**Comment by puzpuzpuz** [comment]
@bziobrowski could you take another look at this one? I addressed your comment and also fixed an unrelated flaky test (`O3FailureTest`).

**Review by bziobrowski — APPROVED** [review]

**Comment by puzpuzpuz** [comment]
@bziobrowski thanks for the review!

**Pull request merged**